### PR TITLE
Add support for Statistics endpoint.

### PR DESCRIPTION
**Pull request by tmcgilchrist**
See https://github.com/mirage/ocaml-github/issues/86 and https://developer.github.com/v3/repos/statistics/.

### CODE REVIEW

**Comment by Aaylor** [comment]
I have an issue while testing the weekly commit count API function: an error occurs during the parsing:

```
Testing weekly commit count.
Fatal error: exception (Failure
   "Bad response: Yojson.Json_error(\"Line 1, bytes 0-33:\\nExpected '{' but found '[[1329609600,5896,-6],[1330214400'\")\
  \nJSON body:\
  \n[\
  \n  [ 1329609600, 5896, -6 ],\
  \n  [ 1330214400, 1848, -1339 ],\
  \n  [ 1330819200, 0, 0 ],\
  [....]
  \n  [ 1598140800, 0, 0 ],\
  \n  [ 1598745600, 39, -5 ],\
  \n  [ 1599350400, 0, 0 ]\
  \n]")
```

Here is the snippet I used to test the function:
```
let test_weekly_commit_count =
  let token = ... in
  let user = "mirage" in
  let repo = "ocaml-github" in
  let open Github in
  Format.printf "Testing weekly commit count.@.";
  let weekly_count =
    let t =
      let open Monad in
      run (Stats.weekly_commit_count ~token ~user ~repo () >|= Response.value)
    in
    Lwt_main.run t
  in
  let pp_sep fmt () = Format.fprintf fmt "; " in
  Format.printf " all: [ %a ]@."
    (Format.pp_print_list ~pp_sep Format.pp_print_int)
    weekly_count.Github_t.participation_all;
  Format.printf " owner: [ %a ]@."
    (Format.pp_print_list ~pp_sep Format.pp_print_int)
    weekly_count.Github_t.participation_owner
```

File: lib/github_core.ml [hunk 1]
```diff
@@ -2002,6 +2011,45 @@ module Make(Env : Github_s.Env)(Time : Github_s.Time)(CL : Cohttp_lwt.S.Client)
...
+          API.code_handler
+            ~expected_code:`Accepted
+            (fun _ -> Lwt.return [])
+        ] in
+      API.get_stream ?token ~uri
+        ~fail_handlers
+        (fun b -> return (code_frequencies_of_string b))
+
+    let weekly_commit_count ?token ~user ~repo () =
+      let uri = URI.repo_code_frequency_stats ~user ~repo in
```
Comment (Aaylor) [inline lib/github_core.ml:2038]: I bet this is the issue causing the parse error: you are not using the correct URI for the `weekly_commit_count`. It should be `/repos/:owner/:repo/stats/participation`, not `/repos/:owner/:repo/stats/code_frequency`!

Comment (avsm) [reply lib/github_core.ml:2038]: Good catch!

Comment (tmcgilchrist) [reply lib/github_core.ml]: Yes of course that's right. I'll add another test program to exercise all the stats endpoints I've added.

**Comment by tmcgilchrist** [comment]
I've added a rudimentary executable that pokes at `mirage/ocaml-github` and dumps a partially formatted representation to stdout. Is there anything else you'd like to see added @Aaylor @avsm ?

**Comment by Aaylor** [comment]
Thanks, it looks good to me; I've run succesfully the different endpoints.

**Pull request merged**